### PR TITLE
METRON-1071: Create CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by justinleet**
## Contributor Comments
Totally forgot I'd been sitting on a half done version of this.  It's basically a stripped down version of the Development Guidelines to make it more obvious how to commit.  Honestly, I'd rather help people out in the contributing phase than for the initial ask to be seeing if our dev guidelines can be read faster than Les Miserables.

Let me know how we want to adjust this, if this is the right level of detail, etc.

## Pull Request Checklist

Thank you for submitting a contribution to Apache Metron.  
Please refer to our [Development Guidelines](https://cwiki.apache.org/confluence/pages/viewpage.action?pageId=61332235) for the complete guide to follow for contributions.  
Please refer also to our [Build Verification Guidelines](https://cwiki.apache.org/confluence/display/METRON/Verifying+Builds?show-miniview) for complete smoke testing guides.  

In order to streamline the review of the contribution we ask you follow these guidelines and ask you to double check the following:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? If not one needs to be created at [Metron Jira](https://issues.apache.org/jira/browse/METRON/?selectedTab=com.atlassian.jira.jira-projects-plugin:summary-panel). 
- [x] Does your PR title start with METRON-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.
- [x] Has your PR been rebased against the latest commit within the target branch (typically master)?

### For documentation related changes:
- [x] Have you ensured that format looks appropriate for the output in which it is rendered by building and verifying the site-book? If not then run the following commands and the verify changes via `site-book/target/site/index.html`:

  ```
  cd site-book
  mvn site
  ```

#### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.
It is also recommended that [travis-ci](https://travis-ci.org) is set up for your personal repository such that your branches are built there before submitting a pull request.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
+#  How To Contribute
+As an open source project, Metron welcomes contributions of all forms. There are several great ways to contribute!
+
+* [Contributing a Code Change](#contributing-a-code-change)
+* Reviewing pull requests on our GitHub page. Check out current open [Pull Requests](https://github.com/apache/metron/pulls)
+* Improve our documentation. Our docs are self contained in the project in README files. Doc changes is the same process as a code change. See [Contributing a Code Change](#contributing-a-code-change)
```
Comment (JonZeolla) [inline CONTRIBUTING.md]: `s/is/are/`?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+* Contributing to or starting discussions on the mailing lists. Both the user and dev lists are great places to give and receive help, or provide feedback.
```
Comment (JonZeolla) [inline CONTRIBUTING.md]: We should point people to the lists.  I'm mobile so not sure what the anchor is on here http://metron.apache.org/community

File: CONTRIBUTING.md (hunk 1 above)
Comment (justinleet) [reply CONTRIBUTING.md]: Good catch.  Changed it.

File: CONTRIBUTING.md (hunk 2 above)
Comment (justinleet) [reply CONTRIBUTING.md]: Added the link to the site.

**Comment by ottobackwards** [comment]
+1, although this makes me wonder again about apache lic. in MD

**Comment by justinleet** [comment]
@ottobackwards I think you're right, we should have the header in the MD files.  I looked back at the discussion you kicked off, the general agreement was that we should, but we didn't follow up.

I've added it here and validated that it doesn't show up in the rendering on GitHub or in the site-book. I'll be creating a Jira and PR to add it to the remaining markdown files, and removing the exclusion from the rat.

@JonZeolla You okay with this change?

**Comment by JonZeolla** [comment]
Yup, looks good

**Pull request closed without merging**